integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/timeseries/TestTimeSeriesCreateTable.scala

```diff
@@ -82,149 +92,209 @@ class TestTimeSeriesCreateTable extends QueryTest with BeforeAndAfterAll {
-           | AS SELECT dataTime, SUM(age) FROM mainTable
-           | GROUP BY dataTime
-       """.stripMargin)
+           | 'EVENT_TIME'='dataTime',
+           | 'HOUR_GRANULARITY'='1')
+           | AS SELECT name, SUM(age) FROM mainTable
+           | GROUP BY name
+         """.stripMargin)
+    }
+    assert(e.getMessage.equals("Time series column dataTime does not exists in select"))
+  }
+
+  test("test timeseries create table 13: don't support create timeseries table on non timestamp") {
+    sql("DROP DATAMAP IF EXISTS agg0_hour ON TABLE mainTable")
+    val e = intercept[MalformedCarbonCommandException] {
+      sql(
+        s"""CREATE DATAMAP agg0_hour ON TABLE mainTable
+           |USING '$timeSeries'
+           |DMPROPERTIES (
+           |   'event_time'='name',
+           |   'HOUR_GRANULARITY'='1')
+           |AS SELECT name, SUM(age) FROM mainTable
+           |GROUP BY name
+        """.stripMargin)
     }
+    assert(e.getMessage.contains("Timeseries event time is only supported on Timestamp column"))
   }
 
-  test("test timeseries create table 11: USING") {
+  test("test timeseries create table 14: USING") {
     val e: Exception = intercept[MalformedDataMapCommandException] {
       sql(
-        """CREATE DATAMAP agg1 ON TABLE mainTable
+        """CREATE DATAMAP agg0_hour ON TABLE mainTable
           | USING 'abc'
           | DMPROPERTIES (
           |   'EVENT_TIME'='dataTime',
-          |   'SECOND_GRANULARITY'='1')
+          |   'HOUR_GRANULARITY'='1')
           | AS SELECT dataTime, SUM(age) FROM mainTable
           | GROUP BY dataTime
         """.stripMargin)
     }
     assert(e.getMessage.equals("DataMap 'abc' not found"))
   }
 
-  test("test timeseries create table 12: USING and catch MalformedCarbonCommandException") {
+  test("test timeseries create table 15: USING and catch MalformedCarbonCommandException") {
     val e: Exception = intercept[MalformedCarbonCommandException] {
       sql(
-        """CREATE DATAMAP agg1 ON TABLE mainTable
+        """CREATE DATAMAP agg0_hour ON TABLE mainTable
           | USING 'abc'
           | DMPROPERTIES (
           |   'EVENT_TIME'='dataTime',
-          |   'SECOND_GRANULARITY'='1')
+          |   'HOUR_GRANULARITY'='1')
           | AS SELECT dataTime, SUM(age) FROM mainTable
           | GROUP BY dataTime
         """.stripMargin)
     }
     assert(e.getMessage.equals("DataMap 'abc' not found"))
   }
 
-  test("test timeseries create table 13: Only one granularity level can be defined 1") {
-    sql("DROP DATAMAP IF EXISTS agg0_second ON TABLE mainTable")
+  test("test timeseries create table 16: Only one granularity level can be defined 1") {
+    sql("DROP DATAMAP IF EXISTS agg0_hour ON TABLE mainTable")
     val e: Exception = intercept[MalformedCarbonCommandException] {
       sql(
         s"""
-           | CREATE DATAMAP agg0_second ON TABLE mainTable
+           | CREATE DATAMAP agg0_hour ON TABLE mainTable
```

Comment:
why change just the name of the table??

Reply:
It's unreasonable to create pre-aggregate with second granularity, second aggregate is used less than hour in reality life.